### PR TITLE
Wallet connect meta data

### PR DESCRIPTION
**Pull request by anxolin**
First PR to better handle Wallet Connect.

## Motivation
This PR adds a new hook which aims to simplify the wallet logic. 

The initial motivation was to have information about what is the connected wallet (not just the provider, but the actual wallet). This will be base to progressively enable wallets depending on their support.

## Scope
However the scope of this PR is slightly bigger. It became evident that the wallet logic is a bit complex, and there's many conditional rendering depending on introspection in the providers, and connected wallet information (see for example Web3StatusInner, Web3StatusInner, etc..)

This PR tries to create a simple interface that would give us all relevant wallet information in something closer to what the react component would want to use. For this, a new hook `useWalletInfo` is implemented.

This hook exposes the following information:
```ts
export interface ConnectedWalletInfo {
  active: boolean
  account?: string | null
  activeNetwork: boolean // active default connection
  provider?: WalletProvider
  isSmartContractWallet: boolean
  walletName?: string
  ensName?: string
  icon?: string
  isSupportedWallet: boolean
}
```

Not all these props will be implemented in this PR, but it's the goal to define a basic abstraction of the wallet, implement the basic ones. 

So `isSupportedWallet` and `isSmartContractWallet` are just place holders for futur PRs.

The most relevant part in this PR will be `walletName`, which will show you the wallet name if available. This comes from the wallet connect peer meta-info that wallets provide.

The icon for now only contains information of the same metada of wallet connect, however, we could have some famous wallets icons and resolve the image by name of the wallet. i.e. Argent don't send the icon, but send the wallet name

### Connect with Metamask
Connect with metamask and observe the info provided by the hook in the log:
image

### Connect with WC using different wallets
image

### CODE REVIEW

File: src/custom/components/Web3Status/index.tsx [hunk 1]
```diff
@@ -1,22 +1,18 @@
 import React, { useMemo } from 'react'
-import { useWeb3React } from '@web3-react/core'
-import useENSName from 'hooks/useENSName'
-import { NetworkContextName } from 'constants/index'
 
 import WalletModal from 'components/WalletModal'
 import { Web3StatusInner } from './Web3StatusMod'
 import useRecentActivity, { TransactionAndOrder } from 'hooks/useRecentActivity'
 import { OrderStatus } from 'state/orders/actions'
+import { useWalletInfo } from '@src/custom/hooks/useWalletInfo'
```
Comment (W3stside) [inline src/custom/components/Web3Status/index.tsx:7]: path

**Review by W3stside — APPROVED** [review]
i like this

**Review by alfetopito — APPROVED** [review]
Working good!

Tested with
- TrustWallet: able to place order
- Argent: able to go as far as enabling the sell token :)
Of course it failed, but I noticed it does not return the `walletName`, the only data set is `ensName`

**Comment by alfetopito** [comment]
Re-tested with all supported and unsupported wallets that I currently have access to, which are:
## Smart contract
1. Gnosis Safe - unsupported :heavy_check_mark: 
2. Argent - unsupported :heavy_check_mark: 
3. Pillar - unsupported :heavy_check_mark: 
## Don't support signature
4. Crypto.com Defi Wallet - unsupported :heavy_check_mark: 
1. TokenPocket - unsupported :heavy_check_mark: 
## Working fine
6. Metamask Mobile - supported :heavy_check_mark: 
1. TrustWallet - supported :heavy_check_mark: 
1. imToken  - supported :heavy_check_mark:

**Pull request merged**